scripts/globals/promyvion.lua

```diff
@@ -0,0 +1,153 @@
+require("scripts/zones/Promyvion-Dem/IDs")
+require("scripts/zones/Promyvion-Holla/IDs")
+require("scripts/zones/Promyvion-Mea/IDs")
+require("scripts/zones/Promyvion-Vahzl/IDs")
+require("scripts/globals/status")
+------------------------------------
+
+dsp = dsp or {}
+dsp.promyvion = dsp.promyvion or {}
+
+------------------------------------
+-- LOCAL FUNCTIONS
+------------------------------------
+
+local function maxFloor(ID)
+    local m = 0
+
+    for _, v in pairs(ID.mob.MEMORY_RECEPTACLES) do
+        m = math.max(m, v[1])
+    end
+
+    return m
+end
+
+local function randomizeFloorExit(ID, floor)
+    local streams = {}
+
+    for _, v in pairs(ID.mob.MEMORY_RECEPTACLES) do
+        if v[1] == floor then
+            GetNPCByID(v[3]):setLocalVar("[promy]floorExit", 0)
+            table.insert(streams, v[3])
+        end
+    end
+
+    local exitStreamId = streams[math.random(#streams)]
+    GetNPCByID(exitStreamId):setLocalVar("[promy]floorExit", 1)
+end
+
+local function findMother(mob)
+    local ID = zones[mob:getZoneID()]
+    local mobId = mob:getID()
+    local mother = 0
+    for k, v in pairs(ID.mob.MEMORY_RECEPTACLES) do
+        if k < mobId and k > mother then
+            mother = k
+        end
+    end
+    return mother
+end
+
+local function checkStray(mob)
+    local ID = zones[mob:getZoneID()]
+    local mobId = mob:getID()
+    local numStrays = ID.mob.MEMORY_RECEPTACLES[mobId][2]
+
+    if os.time() > mob:getLocalVar("[promy]nextStray") then
+        for i = mobId + 1, mobId + numStrays do
+            local stray = GetMobByID(i)
+            if not stray:isSpawned() then
+                mob:setLocalVar("[promy]nextStray", os.time() + 20)
+                SpawnMob(stray:getID())
+                break
+            end
+        end
+    else
+        mob:AnimationSub(2)
+    end
+end
+
+------------------------------------
+-- PUBLIC FUNCTIONS
+------------------------------------
+
+dsp.promyvion.initZone = function(zone)
+    local ID = zones[zone:getID()]
+
+    -- register teleporter regions
+    for k, v in pairs(ID.npc.MEMORY_STREAMS) do
+        zone:registerRegion(k,v[1],v[2],v[3],v[4],v[5],v[6])
+    end
+
+    -- randomize floor exits
+    for i = 1, maxFloor(ID) do
+        randomizeFloorExit(ID, i)
+    end
+end
+
+dsp.promyvion.strayOnSpawn = function(mob)
+    local mother = GetMobByID(findMother(mob))
+
+    if mother ~= nil and mother:isSpawned() then
+        mob:setPos(mother:getXPos(), mother:getYPos() - 5, mother:getZPos())
+        mother:AnimationSub(1)
+    end
+end
+
+dsp.promyvion.receptacleOnFight = function(mob, target)
+    local ID = zones[mob:getZoneID()]
+    local mobId = mob:getID()
+    local numStrays = ID.mob.MEMORY_RECEPTACLES[mobId][2]
+
+    -- keep pets linked
```

Comment:
They do start off with initial agro on whoever has main hate on the receptacle at spawn time, if it is engaged. Otherwise they just hang around waiting for somebody to look at them wrong and if they are already fighting somebody they don't change target.